util/dvsim/NcLauncher.py

```diff
@@ -150,14 +157,51 @@ def poll(self):
         """
 
         assert self.process is not None
-        elapsed_time = datetime.datetime.now() - self.start_time
-        job_runtime_secs = elapsed_time.total_seconds()
         if self.process.poll() is None:
-            timeout_mins = self.deploy.get_timeout_mins()
-            if timeout_mins is not None and not self.deploy.gui:
-                if job_runtime_secs > (timeout_mins * 60):
+            run_timeout_mins = self.deploy.get_timeout_mins()
+            wait_timeout_mins = 180  # max wait time in job queue / license
+            file_size_thresh_bytes = 5000  # log file size threshold
+            if run_timeout_mins is not None and not self.deploy.gui:
```

Comment:
I'm not sure about the gui term here. I wouldn't be happy with waiting 3 hours for the window to come up anyway :-)

Reply:
I am not sure how my changes affect the gui mode as the 'if' statement is only entered when the gui mode is not set (and run_time_mins is set)
The problem with the original implementation was some tests would define run_time_mins as small as 1min and the polling logic didn't account for resource wait times
I have moved the setting of wait_timeout_mins and file_size_thresh_bytes and the associated comment inside the 'if' block